File: C++/two-sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {

        if (nums.size() == 0)
            return vector<int>{};

        for (int i = 0; i < nums.size(); i++)
        {
            for (int j = 0; j < i; j++)
            {
                int sum = nums[i] + nums[j];
                if (sum == target)
                {
                    return vector<int>{i, j};
                }
            }
        }

        return vector<int>{};
    }
};

File: C#/TwoSum.cs
//O(N)
//O(N)
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {

        if (nums.Length == 0) return new int[] { };

        var dict = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int result = target - nums[i];

            if (dict.ContainsKey(result))
            {
                return new int[] { dict[result], i };
            }

            dict[nums[i]] = i;
        }

        return new int[] { };
    }
}